### PR TITLE
WA-21534: Manageable randomiser for unit-tests

### PR DESCRIPTION
**Pull request by skobelevis**
## Summary
Added check for service if it is in map to shuffle tests

### CODE REVIEW

File: suite/suite.go [hunk 1]
```diff
@@ -256,3 +269,21 @@ func runTests(t testing.TB, tests []testing.InternalTest) {
 type runner interface {
 	Run(name string, f func(t *testing.T)) bool
 }
+
+func cleanPath(str string) string {
```
Comment (tanel) [inline suite/suite.go:273]: maybe lets use "path" instead of "str" ?

File: suite/suite.go [hunk 2]
```diff
@@ -256,3 +269,21 @@ func runTests(t testing.TB, tests []testing.InternalTest) {
...
+	replaced := strings.Replace(str, monorepoPath+"/", "", -1)
+
+	r := strings.Split(replaced, "/")
```
Comment (tanel) [inline suite/suite.go:276]: what is "r"? can we give it a better name?

File: suite/suite.go [hunk 3]
```diff
@@ -256,3 +269,21 @@ func runTests(t testing.TB, tests []testing.InternalTest) {
...
+	if r[0] == "pkg" {
+		return strings.Split(str, "/")[1]
```
Comment (tanel) [inline suite/suite.go:278]: can we please add a comment here, why/whats happening

File: suite/suite.go [hunk 4]
```diff
@@ -256,3 +269,21 @@ func runTests(t testing.TB, tests []testing.InternalTest) {
...
+	}
+
+	return r[0]
+}
+
+var monorepoPath = path.Join(os.Getenv("GOPATH"), "src", "github.com", "wallester", "monorepo")
+
+var servicesToShuffle = map[string]struct{}{
+	"card-reference-numbers-management-service": {},
```
Comment (tanel) [inline suite/suite.go:287]: lets add a comment somewhere, why only these 2 are okay to shuffle at the moment

**Review by Djenchik — APPROVED** [review]

**Review by nikitaratnikov — APPROVED** [review]

**Pull request merged**